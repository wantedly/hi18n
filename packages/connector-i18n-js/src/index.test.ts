import { describe, expect, it } from "vitest";
import path from "node:path";
import { connector } from "./index.js";

describe("importData", () => {
  it("imports data from config/locales", async () => {
    const { importData } = connector(
      path.resolve(__dirname, "./__fixtures__/project1/input/.hi18nrc.json"),
      {}
    );
    const data = await importData!();
    expect(data).toMatchSnapshot();
  });
});
