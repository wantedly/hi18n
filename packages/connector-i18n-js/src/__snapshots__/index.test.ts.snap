// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`importData > imports data from config/locales 1`] = `
{
  "translations": {
    "en": {
      "messages.greeting": {
        "raw": "Hello, {name}!",
      },
      "messages.unread": {
        "raw": "{count, plural, one {You have # unread message.} other {You have # unread messages.}}",
      },
      "messages.unread.one": {
        "raw": "You have {count} unread message.",
      },
      "messages.unread.other": {
        "raw": "You have {count} unread messages.",
      },
    },
    "ja": {
      "messages.greeting": {
        "raw": "こんにちは、{name}さん!",
      },
      "messages.unread": {
        "raw": "未読のメッセージが{count}件あります",
      },
    },
  },
}
`;
